keanu-project/src/main/java/io/improbable/keanu/util/io/ProtobufLoader.java

```diff
@@ -211,10 +213,17 @@ private Vertex instantiateVertex(Class vertexClass,
         Object[] arguments = new Object[constructorParameters.length];
 
         for (int i = 0; i < constructorParameters.length; i++) {
-            arguments[i] = getParameter(constructorParameters[i], paramMap, vertex);
+            Object parameter = getParameter(constructorParameters[i], paramMap, vertex);
+            arguments[i] = parameter;
 
-            Class argumentClass = arguments[i].getClass();
-            Class parameterClass = Primitives.wrap(constructorParameters[i].getType());
+            Class argumentClass;
+            Class parameterClass;
+            if (parameter != null) {
+                argumentClass = arguments[i].getClass();
+            } else {
+                argumentClass = constructorParameters[i].getType();
```

Comment:
if we init parameterClass before this if block, we can just use it here instead of doing constructorParameters... again.